rest/oidc_api.go

```diff
@@ -237,6 +237,14 @@ func (h *handler) getOIDCCallbackURL() string {
 		base.Warnf("Can't calculate OIDC callback URL without DB in path.")
 		return ""
 	} else {
-		return fmt.Sprintf("%s://%s/%s/%s", scheme, h.rq.Host, dbName, "_oidc_callback")
+		callbackURL := fmt.Sprintf("%s://%s/%s/_oidc_callback", scheme, h.rq.Host, dbName)
+		if !isDefault && providerName != "" {
+			callbackURL, err := auth.SetURLQueryParam(callbackURL, auth.OIDCAuthProvider, providerName)
+			if err != nil {
+				base.Warnf("Failed to add provider %q to OIDC callback URL (%s): %v", base.UD(providerName), callbackURL, err)
+			}
+			return callbackURL
+		}
+		return callbackURL
 	}
```

Comment:
I'm not a fan of the nesting in this function. I know it was originally like this, but it seems like it can be cleaned up a bit to prevent future scoping issues.

```go
	dbName := h.PathVar("db")
	if dbName == "" {
		base.Warnf("Can't calculate OIDC callback URL without DB in path.")
		return ""
	}

	scheme := "http"
	if h.rq.TLS != nil {
		scheme = "https"
	}

	callbackURL := scheme + "://" + h.rq.Host + "/" + dbName + "/_oidc_callback"
	if isDefault || providerName == "" {
		return callbackURL
	}

	callbackURL, err := auth.SetURLQueryParam(callbackURL, auth.OIDCAuthProvider, providerName)
	if err != nil {
		base.Warnf("Failed to add provider %q to OIDC callback URL (%s): %v", base.UD(providerName), callbackURL, err)
	}
	return callbackURL
```

Reply:
Revised getOIDCCallbackURL function to avoid nesting inside it.